src/lib/commandclass/ColorSwitchCC.test.ts

```diff
@@ -0,0 +1,265 @@
+import { createEmptyMockDriver } from "../../../test/mocks";
+import type { Driver } from "../driver/Driver";
+import { ZWaveNode } from "../node/Node";
+import { Duration } from "../values/Duration";
+import {
+	ColorComponent,
+	ColorSwitchCCGet,
+	ColorSwitchCCReport,
+	ColorSwitchCCSet,
+	ColorSwitchCCStartLevelChange,
+	ColorSwitchCCStopLevelChange,
+	ColorSwitchCCSupportedGet,
+	ColorSwitchCCSupportedReport,
+	ColorSwitchCommand,
+} from "./ColorSwitchCC";
+import { CommandClasses } from "./CommandClasses";
+
+const fakeDriver = (createEmptyMockDriver() as unknown) as Driver;
+const node1 = new ZWaveNode(1, fakeDriver as any);
+(fakeDriver.controller!.nodes as any).set(1, node1);
+
+function buildCCBuffer(payload: Buffer): Buffer {
+	return Buffer.concat([
+		Buffer.from([
+			CommandClasses["Color Switch"], // CC
+		]),
+		payload,
+	]);
+}
+
+describe.only("lib/commandclass/ColorSwitchCC => ", () => {
+	it("the SupportedGet command should serialize correctly", () => {
+		const cc = new ColorSwitchCCSupportedGet(fakeDriver, {
+			nodeId: 1,
+		});
+		const expected = buildCCBuffer(
+			Buffer.from([
+				ColorSwitchCommand.SupportedGet, // CC Command
+			]),
+		);
+		expect(cc.serialize()).toEqual(expected);
+	});
+
+	it("the SupportedReport command should deserialize correctly", () => {
+		const ccData = buildCCBuffer(
+			Buffer.from([
+				ColorSwitchCommand.SupportedReport, // CC Command
+				0b0001_1111,
+				0b0000_0001,
+			]),
+		);
+		const cc = new ColorSwitchCCSupportedReport(fakeDriver, {
+			nodeId: 1,
+			data: ccData,
+		});
+
+		expect(cc.supportsColdWhite).toBe(true);
+		expect(cc.supportsWarmWhite).toBe(true);
+		expect(cc.supportsRed).toBe(true);
+		expect(cc.supportsGreen).toBe(true);
+		expect(cc.supportsBlue).toBe(true);
+		expect(cc.supportsAmber).toBe(false);
+		expect(cc.supportsCyan).toBe(false);
+		expect(cc.supportsPurple).toBe(false);
+		expect(cc.supportsIndex).toBe(true);
+	});
+
+	it("the Get command should serialize correctly", () => {
+		const cc = new ColorSwitchCCGet(fakeDriver, {
+			nodeId: 1,
+			colorComponent: ColorComponent.Red,
+		});
+		const expected = buildCCBuffer(
+			Buffer.from([
+				ColorSwitchCommand.Get, // CC Command
+				2, // Color Component
+			]),
+		);
+		expect(cc.serialize()).toEqual(expected);
+	});
+
+	// TODO: How do we test old versions of deserialization?
+	/*
+	it("the Report command should deserialize correctly (version 1)", () => {
+		const ccData = buildCCBuffer(
+			Buffer.from([
+				ColorSwitchCommand.Report, // CC Command
+				0b0000_0010, // color: red
+				0b1111_1111, // value: 255
+			]),
+		);
+		const cc = new ColorSwitchCCReport(fakeDriver, {
+      nodeId: 1,
+			data: ccData,
+		});
+		cc.version = 1; // Will not work, as deserialization is performed in the ctor
+
```

Comment:
I forgot to comment here. Deserialization should usually be independent of the version, since you cannot be sure that the version is known before receiving the command. What exactly is the problem here?

Reply:
I was trying to test that the deserialization for older versions works correctly.  I couldn't find a way to pre-set the version so I disabled the entire test.  I am actually confused now as to how the v3 test is working.  Something must be prepopulating the version property to something >= 3 for that test to be passing.

Reply:
> Something must be prepopulating the version property to something >= 3 for that test to be passing.

Yeah. The mock driver used for tests is setting the version to 100. To test the deserialization for e.g. v1, you should just give the method a buffer that is valid in v1 (usually has less fields).

Reply:
I am checking the version field to determine what I should parse though.  In order to make the test work like that, I would have to ignore the version and ,make the cc parser rely only on the length of the payload.  Is this really what we want to do in this case?

Reply:
> I would have to ignore the version and ,make the cc parser rely only on the length of the payload. Is this really what we want to do in this case?

I'll need to take a look at the specifics, but this is actually what the specs say about this.